src/structs/probability_tables.rs

```diff
@@ -218,22 +216,34 @@ impl ProbabilityTables {
             };
         }
 
-        let mut best_prior = compute_lak_coeffs_a[0] * coef_idct[coef_idct_offset]; // rounding towards zero before adding coeffs_a[0] helps ratio slightly, but this is cheaper
-        for i in 1..8 {
-            let sign = if (i & 1) != 0 { 1 } else { -1 };
-            best_prior -= coef_idct[coef_idct_offset + i]
-                * (compute_lak_coeffs_x[i] + (sign * compute_lak_coeffs_a[i]));
+        let mut best_prior = 0;
+        for i in 0..8 {
+            best_prior += coef_idct[i] * (compute_lak_coeffs_a[i] - compute_lak_coeffs_x[i]);
+            // rounding towards zero before adding coeffs_a[0] helps ratio slightly, but this is cheaper
         }
 
-        best_prior /= coef_idct[coef_idct_offset];
+        best_prior /= coef_idct[0];
 
         return ProbabilityTablesCoefficientContext {
             best_prior,
             num_non_zeros_bin: num_non_zeros_x,
-            best_prior_bit_len: u32_bit_length(cmp::min(best_prior.abs() as u32, 1023)),
+            best_prior_bit_len: u32_bit_length(cmp::min(best_prior.unsigned_abs(), 1023)),
         };
     }
 
+    fn from_stride(block: &[i16; 64], offset: usize, stride: usize) -> i16x8 {
+        return i16x8::new([
```

Comment:
can't wait until they stabilize portable_simd
I still remember that they stabilized it once--then unstabilized it a few days later---that was a sad day--I think one of the very few rust features that snuck in for a moment
I still have programs that worked with stable and then stopped working